interfaces/cython/cantera/onedim.py

```diff
@@ -34,6 +28,41 @@ def __init__(self, domains, gas, grid=None):
         self.gas = gas
         self.flame.P = gas.P
 
+    def extra(self, domain=None):
+        """
+        Extra columns used for saving/restoring of simulation data that are
+        specific to a class derived from `FlameBase`. Entries may include:
+         * ``grid``: grid point positions along the flame [m]
+         * ``velocity``: normal velocity [m/s]
+         * ``spread_rate``: tangential velocity gradient [1/s]
+         * ``lambda``: radial pressure gradient [N/m^4]
+         * ``eField``: electric field strength
+        """
+        if domain is None:
+            return self._extra
+
+        dom = self.domains[self.domain_index(domain)]
+        if isinstance(dom, Inlet1D):
+            return tuple([e for e in self._extra
+                          if e not in {'grid', 'lambda', 'eField'}])
+        elif isinstance(dom, IdealGasFlow):
+            return self._extra
+        else:
+            return ()
+
+    def phase(self, domain=None):
+        """
+        Return phase describing the domain (i.e. gas phase or surface phase).
+        """
```

Comment:
Well, it's actually used to access domains within the `Sim1D.domains` list. It can be relocated there (which I did); also, `phase` can be defined within `Domain1D` and `ReactingSurf1D` as you suggested (which I also implemented).